src/main/java/net/dv8tion/jda/internal/entities/PrivateChannelImpl.java

```diff
@@ -46,26 +48,42 @@ public ChannelType getType()
         return ChannelType.PRIVATE;
     }
 
-    @Nonnull
+    @Nullable
     @Override
     public User getUser()
     {
         updateUser();
         return user;
     }
 
+    @Nonnull
+    @Override
+    public RestAction<User> retrieveUser()
+    {
+        if (user != null)
+            return new CompletedRestAction<>(getJDA(), user);
+        //even if the user blocks the bot, this does not fail.
+        return retrievePrivateChannel()
+                .map(PrivateChannel::getUser);
+    }
+
     @Nonnull
     @Override
     public String getName()
     {
+        if (getUser() == null)
```

Comment:
I would make this getUser() save to a local variable, since every call to getUser has to acquire the cache read-lock.